Improve documentation (#86)
vCloud-DFTBA/faythe: docs/getting-started.md

- `basic_auth`: HTTP basic authentication with `username` & `password`. If you don't want to enable basic authentication, just remove this section.

### 3.2. Etcd configuration

Etcd is a Faythe's heart. You can figure out etcd configuration [here](https://github.com/etcd-io/etcd/blob/master/clientv3/config.go). The configuration in Faythe is just a mimic.

## 4. Terms

There are the illustration about the Faythe objects:

![](./imgs/objects.png)

Every Faythe objects is stored in Etcd.

### 4.1. Cloud provider

- Etcd path, the cloud provider id which is created by hash its auth-url.

```
/clouds/<provider-id>
```

- The main object, represents Cloud information: type, auth url, credentials, its monitor engine...

- It contains the rest of objects (scalers, healers...) logically.

- The only supported provider type at this time is [OpenStack](https://openstack.org/).

### 4.2. Scaler

- Etcd path:

```
/scalers/<provider-id>/<scaler-id>
```

- `Scaler` has a user-defined query which can be evaluated based on Monitor engine metrics then execute the defined actions (HTTP Action to scale up/scale down). The query must be follow the Monitor Engine query format.

> - One `Cloud provider` can have many `Scaler`.

Reviewer: ```suggestion
- One `Cloud provider` can have **many** `Scaler`.
```

### 4.3. Healer

- Etcd path:

```
/healers/<provider-id>/<healer-id>
```

- `Healer` receives the query from user and evaluate the need of healing for hosts based on that query. The query must be follow the Monitor Engine query format.

> - One `Cloud provider` has only one `Healer`.

Reviewer: ```suggestion
- One `Cloud provider` has **only one** `Healer`.
```

### 4.4. NResolver (Name resolver)

- Etcd path:

```
/nresolvers/<provider-id>/<nresolver-id>
```

- Name resolver as the name it tells, it resolve host IP address to host's name.

- Some cloud providers, OpenStack for example, execute commands based on host's name not the IP address. On the other hand, some metrics backends only care about host IP address. Hence, Faythe needs to store the mapping of host IP address and its name.

> - One `Cloud provider` has only one `NResolver`.

Reviewer: ```suggestion
- One `Cloud provider` has **only one** `NResolver`.
```

### 4.5. Silencers

- Etcd path:

```
/silencers/<cloud-d>/<silencer-id>
```

- Silencers come in handy if you want to add a set of ignored hosts in case of maintenance. It controls the healer.

> - One `Cloud provider` can have many `Silencers`.

Reviewer: ```suggestion
- One `Cloud provider` can have **many** `Silencers`.
```

### 4.6. Monitor Engine

- A `Cloud provider` always associate with a `Monitor Engine`.

- A `Monitor Engine` represents the monitor information: backend type ([Prometheus](https://prometheus.io) is the only supported backend at this time), url address, credentials,...

- Other objects like scaler, healer... integrate with `Monitor Engine` to evaluate.

### 4.7. AT Engine (Automation Engine)

- A `Cloud provider` can associate with a `AT Engine`.

- An `AT Engine`  represents the automation information: type ([StackStorm](https://stackstorm.com/)), credentials, url address...

- An `AT Engine` allows us to define a complex set of actions: evaluate the virtual machines in the case a host went down, check the evaluation progress as well as verify the results, notify the operators in the end.

## 5. Authentication with JWT

Faythe uses authentication mechanism [JSON Web Token](https://jwt.io/). By passing the username and password, users receive a token be stored in cookie (it will be expired after 1 hour).

Before do other magic things, you have to login first.

## 6. Cloud APIs

> The Cloud provider is the main object so you have to register one first. Check out the follow API.

Reviewer: ```suggestion
The Cloud provider is the main object so you have to register one first. Check out the following APIs.
```
